### ISSUE TITLE
Karma Points / Shop / Player Award System

### ISSUE THREAD

**Issue by Peptide90** [space-wizards/space-station-14#7842]
Some SS13 servers have an option to award karma to other players either during the round or encouraging it at round end.

This is a way for players to recognise other players for good roleplay, badass actions or just general fun. Said karma can then be redeemed on karma locked flavour roles (e.g. musician, barber, pod pilot, mechanic, blueshield, NT rep etc etc), as well as races that have interesting mechanics.

In practice it could also implement well with the load out system, using points for one off next round start bonuses of fluff items and drip or possibly permanent unlocks. The options are endless and highly customisable and I guarantee downstream will be interested.

A lot of players have expressed interest as have some of the team. To moderate the main thing would be putting a cap to make sure players give karma to X amount of other players before they can award to the same player again, as well as allowing admins to check karma logs per player and strip points etc.

**Comment by moonheart08** [comment]
We're not karma locking musician it's perfect as-is. The rest is good though.

**Comment by TimrodDX** [comment]
There's a couple reasons I don't like this idea. 

The first is that it's been tried before, and almost inevitably devolves into two things: the first half is people breaking up into cliques and forming meta-friendships, which is bad for the game and usually results in people being banned, and the second is that most people are just going to pick someone at random because no one remembers anyone's name in SS13. You don't want to encourage either of these things. 

Second, you don't want to lock content behind a wall in a game where there's already a huge problem of people joining and not having anything to do. That's how you get the grey tide. Giving someone a job, any job, is better than having them grey tide - and I think having jobs like that are how you divert people out of grey tiding. 

Third, it cheapens the gameplay value of admin rounds. A lot of that stuff you mention is stuff that was admin-exclusive in /tg/, and was used to make some of the most memorable rounds I ever played through - things like Centcom rounds, Thunderdome, NT rep visits, death squads, that kind of thing. If anyone can do those things, it's a lot less impressive and a lot less memorable. 

I just think it's the wrong way of doing things.

**Comment by ike709** [comment]
Player-awarded Karma™ is a terrible idea for numerous reasons, one of the main ones being metafriends & point-for-point trading. I could go on for a while but this is one of the biggest.

Some form of generic points that the player earns themself (similar to BeeStation's Beecoins™) is less awful but has its own issues (primarily needing sufficient sinks and a cap to stop people from perpetually accumulating an infinite amount).

Or alternatively an admin-awarded benefit of some kind. I know some server gave admins the ability to unlock Ultra job priority (basically High+, being limited to one slot) for specific players, with the idea that better players would be prioritized for (hopefully) the more important roles.

But generally any implementation of this idea has its own problems.

**Comment by Peptide90** [comment]
> There's a couple reasons I don't like this idea.

> But generally any implementation of this idea has its own problems.

I like to play devils advocate so let me raise you two questions then some answers.
**Will it negatively impact you or your round?** and **What's a better way for players to recognise other players for good roleplay or contribution to the round?** and just as an extension, how would you lock up fluff roles and species you don't want being common? (not necessarily event roles).

**The rebuttle**

- Cliques and meta friends happen whether this system exists or not. A lot of our players use the same name between rounds and have hundreds of hours playing together so at this point I'd be concerned if some of them weren't friends in some capacity.
- Moderation steps can be put in place to ensure meta isn't an issue such as the one mentioned in the original post, forcing players to give their points to X amount of OTHER players before they can give it to the same person again. We can have robust logging for checking point transactions too and as we know our players it will be pretty easy to point out meta and just deduct points.
- We aren't forcing people to use their points each round and we can also have limits on place such as the round needs to last for more than 30 minutes or need X amount of players in a round. This will be on our official servers only and downstreams will certainly make use of this to gate races.
- The jobs listed are just some examples however note I did leave out things like thunderdome, centcom and deathsquad because those are event specific. The roles listed are the typical fluff roles which typicall have little effect on the round. The ones that do like blueshield or NT rep can be gated higher and thus require a higher caliber of roleplayer AND we can prioritise other roles first so that they only become available in high pop situations. The content being gated will be for situations when most roles are already full and is not any additional content that can't usually be done in a round anyway, it's often just a title and loadout.
- Having admins award points is arguably worse. It requires admin intervention, we often aren't watching everyone so will miss stuff and some players will generally just never get points.
- And a final question. If roles and races are the issue which ignoring the above, is there an issue with a purely cosmetic loadouts system using the points and further, making most of the cosmetics run out after X rounds or only last for 1 round? This has it's own issues but still, people will have an issue with karma shop and I want some genuine discussion on it.

**Comment by PJB3005** [comment]
> What's a better way for players to recognise other players for good roleplay or contribution to the round?

Expectation: players get awarded for good roleplay

Reality: players get reality for being in the right place at the right time (medbay) or doing a gimmick for 5 consecutive rounds

**Comment by Androidonator** [comment]
I have about few hundred hours on paradise station and I have measly 6 karma I mainly play engi or sec so yeah that might be the reason why. I think goonstation job rewards system is interesting idea but a combination of both would be even better idea in my opinion i.e.

+100 points for getting "commend or whatever you want to call it"
+20 healing
+30 doing the station objective
+10 playtime
+100 points for getting "commend or whatever you want to call it"
+20 healing
+30 doing the station objective
+10 playtime
etc.

I think it shouldn't be purely cosmetic, but it should be either. 

1. minor improvement in stats i.e. 

       - +5 capacity bag or something

3. side-grade (serving same purpose different interesting way) i.e.

      - medical scanner, but it only scans for locations of fractures and organ damage but not for any type of damage.
      - banana peel that is sticky (incremental slowdown as you walk over it) and not slippery

4. convenience (not really game changing, but nice to have) ie..

      - getting zipties instead of handcuffs

**Comment by SlamBamActionman** [comment]
I don't think this feature request holds up on the first examine - at the very least it would need to be thoroughly elaborated on how the system would actually look, because it seems to be as something that could very easily be abused, or end up not mattering. Rewarding players for being "up in your face" with their roleplay can lead to frustration for players who don't want to engage in that type of roleplay; a janitor who cleans the station and have small chats with crew is very valuable to the vibes of the game, possibly more than the hyper passenger running around causing a ruckus in all the departments. 

I'll be closing this feature request due to a lack of current maintainer approval. If there's a desire to pursue this feature, it is recommended to bring it up on the forum/on Discord for further discussion.
